### PR TITLE
Displaying Posts on feed

### LINKED ISSUE #88: Displaying posts on feed
- Displaying post as ListView.
- Updating html and css of the feed.

### PR DESCRIPTION
**Pull request by AlonShakaroff**
- Adding PostListView class to views.
- Displaying Resumes on feed as an embedded PDF.
- Changing the X-FRAME-OPTIONS so that embedded pdf display is allowed and not blocked by the browser.

Dependant on #77  ( please merge #77 before)
fix #88

### CODE REVIEW

**Review by OmerCS8 — APPROVED** [review]
LGTM

**Review by yulisuliman — CHANGES_REQUESTED** [review]
In my opinion each test should be in their app folder and not in a general folder,
So I think that the tests folder that you created should be in posts app.

File: static/styles/main.css [hunk 1]
```diff
@@ -0,0 +1,174 @@
+/* ---------------------------------------------------------- GoogLe Fonts ---------------------------------------------------------- */
+@import url('https://fonts.googleapis.com/css2?family=Oleo+Script+Swash+Caps:wght@400;700&family=Poppins:ital,wght@0,300;0,500;1,600&display=swap');
+/* 
+font-family: 'Oleo Script Swash Caps', cursive;
+font-family: 'Poppins', sans-serif;
```
Comment (TomerNewmanPrograms) [inline static/styles/main.css:5]: Please notice that the comments are here again

**Review by TomerNewmanPrograms — CHANGES_REQUESTED** [review]

File: tests/tests.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import pytest
+
+
+@pytest.mark.django_db
```
Comment (yulisuliman) [inline tests/tests.py:4]: Why you need  @pytest.mark.django_db ?

File: static/styles/main.css (hunk 1 above)
Comment (AlonShakaroff) [reply static/styles/main.css]: Just kicking them out.

File: tests/tests.py (hunk 2 above)
Comment (AlonShakaroff) [reply tests/tests.py:4]: For future use, you can see the usage in the next PR I uploaded.

**Review by MatanP12 — DISMISSED** [review]
LGTM

File: static/styles/main.css [hunk 3]
```diff
@@ -20,10 +20,11 @@ body{
     --gradiant: linear-gradient(to right, #e2336b, #fcac46);
     --background: linear-gradient(180deg, rgb(248, 248, 248) 0%, rgb(255, 255, 255) 100%);
     --lightbackground: linear-gradient( 135deg, #90F7EC 10%, #32CCBC 100%);
+    --pagebackground: #f0f2f5;
 }
 
 .container{
-  width:95%;
+  width:100%;
```
Comment (OmerCS8) [inline static/styles/main.css:27]: Can you add the next line to `about.css` to match these changes?
`width: 80%;` at `.description{... }`

**Review by OmerCS8 — CHANGES_REQUESTED** [review]

File: static/styles/main.css (hunk 3 above)
Comment (AlonShakaroff) [reply static/styles/main.css:27]: Ok

**Comment by AlonShakaroff** [comment]
> In my opinion each test should be in their app folder and not in a general folder, So I think that the tests folder that you created should be in posts app.

Actually the folder already existed, I just added the navigation tests there,
and because it's just the same basic test for all the views (for every view added there should be a navigation test there),
I think it should be in this folder that is not app related but related to all the apps.

**Review by OmerCS8 — APPROVED** [review]
LGTM, great job!

**Review by TomerNewmanPrograms — DISMISSED** [review]
LGTM! :)

File: tests/tests.py (hunk 2 above)
Comment (natifridman) [reply tests/tests.py:4]: If it's not part of this PR, I think you should remove it and add it later

**Comment by natifridman** [comment]
> In my opinion each test should be in their app folder and not in a general folder, So I think that the tests folder that you created should be in posts app.

I tend to agree, you can create multiple test files in the app folder. So instead of having big test files, you can separate them.
For example, you can have test files for the posts view and name it `test_posts_view`.

BTW, I think you can delete the tests folder from the main folder. It only contains the `hello world` test from the pytest setup.

**Comment by AlonShakaroff** [comment]
> e the tests folder from the main folder. It only contains the `hello world` test from the pytest setup.

Ok, I'll do that now.

**Review by MatanP12 — DISMISSED** [review]
LGTM

File: tests/tests.py (hunk 2 above)
Comment (AlonShakaroff) [reply tests/tests.py]: Apparently, it's necessary for the tests that I wrote there I removed that and got an error that explicitly says that it is required to use that annotation for those tests.
@natifridman @yulisuliman

**Comment by natifridman** [comment]
> Ok, I'll do that now. By the way, if I separate the tests into different files, do I need to create fixtures for each one of them separately or can I import fixtures from another test file to avoid code duplication? @natifridman

You can use fixtures from another test file. See this [example](https://docs.pytest.org/en/7.1.x/how-to/fixtures.html#scope-sharing-fixtures-across-classes-modules-packages-or-session) for conftest file.

File: posts/views.py [hunk 4]
```diff
@@ -1,8 +1,13 @@
...
+from .models import Post
 
 
-def feed(request):
-    return render(request, 'posts/feed.html', {'title': 'feed'})
+class PostListView(ListView):
+    model = Post
+    template_name = 'posts/feed.html'
+    context_object_name = 'posts'
+    ordering = ['-date_posted']  # order posts from newest to oldest
```
Comment (OmerCS8) [inline posts/views.py:10]: In order to send the `title` you can add this function after line 10:
def get_context_data(self, **kwargs):
    ctx = super(PostListView, self).get_context_data(**kwargs)
    ctx['title'] = 'feed'
    return ctx
it should look like this:
image

**Review by OmerCS8 — CHANGES_REQUESTED** [review]

File: posts/views.py (hunk 4 above)
Comment (AlonShakaroff) [reply posts/views.py:10]: Done 👍

**Review by OmerCS8 — DISMISSED** [review]
LGTM 👍

**Review by TomerNewmanPrograms — APPROVED** [review]
LGTM

File: posts/templates/posts/feed.html [hunk 5]
```diff
@@ -1,6 +1,23 @@
...
+              <a class="mr-2" href="#">{{ post.author }}</a>
+              <small class="text-muted">{{ post.date_posted|date:"F d, Y" }}</small>
+            </div>
+            <p class="article-content">{{ post.description }}</p>
+              {% if post.resume %}
+                <iframe src="{{ post.resume.resume_file.url }}" style="width: 560px;height: 780px ;border: none;"></iframe>
+              {% endif %}
+          </div>
+        </article>
+        <hr/>
```
Comment (natifridman) [inline posts/templates/posts/feed.html:21]: I don't think you need to close the hr tag
```suggestion
        <hr>
```

**Review by natifridman — CHANGES_REQUESTED** [review]

**Comment by natifridman** [comment]
This is out of this PR scope, but since we have some test files under the posts app, maybe we should create a tests folder for the app and move the tests there?

**Comment by AlonShakaroff** [comment]
> This is out of this PR scope, but since we have some test files under the posts app, maybe we should create a tests folder for the app and move the tests there?

Done 👍

**Review by OmerCS8 — APPROVED** [review]
LGTM

**Review by MatanP12 — DISMISSED** [review]
LGTM

**Review by natifridman — APPROVED** [review]
Looks good, nice work!

**Review by TomerNewmanPrograms — APPROVED** [review]

**Review by yulisuliman — APPROVED** [review]
LGTM

**Review by OmerCS8 — APPROVED** [review]
LGTM

**Review by natifridman — APPROVED** [review]

**Pull request merged**